tests/publiccloud/run_ltp.pm

```diff
@@ -48,6 +47,47 @@ sub instance_log_args
         $self->{provider}->region);
 }
 
+sub upload_ltp_logs
+{
+    my ($self)        = @_;
+    my $log_file      = Mojo::File::path('ulogs/ltp_log.json');
+    my $ltp_testsuite = get_required_var('COMMAND_FILE');
+
+    upload_logs("$root_dir/ltp_log.raw",  log_name => 'ltp_log.raw',       failok => 1);
+    upload_logs("$root_dir/ltp_log.json", log_name => $log_file->basename, failok => 1);
+
+    return unless -e $log_file->to_string;
+
+    local @INC = ($ENV{OPENQA_LIBPATH} // testapi::OPENQA_LIBPATH, @INC);
+    eval {
+        require OpenQA::Parser::Format::LTP;
+
+        my $ltp_log = Mojo::JSON::decode_json($log_file->slurp());
+        my $parser  = OpenQA::Parser::Format::LTP->new()->load($log_file->to_string);
+
+        if (find_whitelist_testsuite($ltp_testsuite)) {
+            for my $result (@{$parser->results()}) {
+                next unless ($result->{result} ne 'ok');
+                my ($ltp_test) = grep { $_->{test_fqn} eq $result->{test_fqn} } @{$ltp_log->{results}};
```

Comment:
It'd be more efficient to turn `$ltp_log->{results}` into a hash. Syscalls testsuite has enough tests to make this O(n^2) array lookup slow if a significant number of them fails.

Reply:
Agree here, the reason for this is "historical". 

We use the `runltp-ng --json_filter=openqa`. This produce a json file which can be consumed by https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Parser/Format/LTP.pm .
From my point of view, we (tests/publiccloud/run_ltp.pm) are the only user of that Parser, thus there is actually no need to have `--json_filter=openqa` anymore, instead adopt the `OpenQA::Parser::Format::LTP` to understand the latest produced output from `runltp-ng` or create a LTP-ng Parser.

But I can add that temp hash...